src/Symfony/Component/Serializer/Encoder/JsonDecode.php

```diff
@@ -78,6 +81,16 @@ public function decode($data, $format, array $context = array())
             throw new NotEncodableValueException(json_last_error_msg());
         }
 
+        if ($rootKey = $context['json_root_key']) {
+            if (\is_array($decodedData) && array_key_exists($rootKey, $decodedData)) {
```

Comment:
I'm not sure we want to support this feature in core, but if we do, I think we should allow to use a property path (e.g: `foo.baz` to access any node) by using the PropertyAccess component.

Reply:
While integrating with different APIs we often face the need in accessing data from certain property, usually root one such as

	{
	    "user": {
		"name": "userName",
		"city": "cityName"
	    }
	} 
So it would be convenient to map it into an object simply by calling the deserialize method.